mava/wrappers/env_preprocess_wrappers.py

```diff
@@ -260,16 +264,94 @@ def obs_normalisation_start_index(self) -> int:
 
         return old_value + num_values
 
-    def __getattr__(self, name: str) -> Any:
-        """Expose any other attributes of the underlying environment.
 
-        Args:
-            name (str): attribute.
+class StackObservations(BasePreprocessWrapper):
+    """Stack frames of observations together"""
 
-        Returns:
-            Any: return attribute from env or underlying env.
-        """
-        if hasattr(self.__class__, name):
-            return self.__getattribute__(name)
+    def __init__(self, environment: Any, num_frames: int = 1):
+        """Initialise wrapper."""
+
+        super().__init__(environment)
+
+        # Check that this is the first wrapper that is called on the environment
+        if self._environment.obs_normalisation_start_index != 0:
+            raise ValueError(
+                "Observation stacking should be the first custom\
+                    wrapper we call in the environment factory."
+            )
+
+        self.frames: Any = {
```

Comment:
I have change that to Dict[str, deque]. I didn't specify any type of the deque and this may vary per environment and observations. Also it will be another overhead to pull it out in the initialisation.